### DISCUSSION TITLE
dynamic kernel whitelist

### DISCUSSION THREAD

**Discussion by bloomsa** [jupyter/enterprise_gateway#1059] [Q&A] [answered]
Hi, I'm wondering about the possibility of dynamically updating the kernels offered by Enterprise Gateway. For example, the Gateway is running in docker with the kernelspecs directory mounted so it can be accessed from outside the container. If I were to place a new kernel (and kernelspec) there, how can I tell the Gateway that there is a new kernel offering, and is it possible to update without restarting the gateway? I know [this env var ](https://github.com/jupyter-server/enterprise_gateway/blob/4fe94a56a34c45d904adb339f2045ed8664b2c43/etc/docker/docker-compose.yml#L26) sets the kernel whitelist and [dynamic configuration](https://jupyter-enterprise-gateway.readthedocs.io/en/latest/operators/config-dynamic.html) is supported for some configs but I was unable to confirm that it works for the kernel whitelist.

**Comment by kevin-bates** [comment, marked as answer]
Hi @bloomsa - thank you for opening this discussion.  To get this behavior, you could simply remove the whitelist env altogether.  By default, new (updated, and deleted) kernelspecs are detected as soon as the next request comes in for them (which I believe tends to be too frequent, but, in this case, is a good thing).

If you wanted to tie this to _dynamic configuration_ you'd need to configure the whitelist via the configuration file which, in container worlds, is also not that helpful.  Command-line and env-based values always override the file-based configurables and, therefore, are not privy to dynamic configuration (which is caveat #1).

**Reply by bloomsa** [reply]
wonderful, thank you for the quick reply! I tried setting the env variable as [‘*’] but not leaving it out entirely. The dynamic config piece of the answer makes sense too. thanks again 🙌

**Comment by kevin-bates** [comment]
For completeness, it turns out the capability discussed here was not practically available for containerized environments until #1131.
